labdao/plex
# Tags

## What type of PR is this? 

- 🎮 Feature

## Description

Tags are added to datafiles upon upload and generation to the gateway. For generated datafiles, they currently have Tags `generated` and `experiment:[experiment-name]`.

[Demo](https://www.loom.com/share/6a77f895a72c4d0d8d3ff75501d24443)

There is also functionality for listing tags, which serves as a basis for filtering datafiles by tag. Accessible via a `GET` request at `/tags`.

## Related Tickets & Documents

https://linear.app/convexitylabs/issue/LAB-801/tag-model-creation
https://linear.app/convexitylabs/issue/LAB-803/generated-tag-for-datafiles
https://linear.app/convexitylabs/issue/LAB-804/filter-datafiles-by-generated

## Steps to Test

1. Upload a datafile.
2. Check that datafile includes `uploaded` tag. This can be seen either by listing tags or directly in the datafile table view.
3. Submit an experiment.
4. Upon completion, check that output datafiles have `generated` and `experiment:[experiment-name]` tags.

## Relevant GIF 

<img src="https://media1.giphy.com/media/l0HlMEi55YsfXyzMk/giphy.gif"/>
